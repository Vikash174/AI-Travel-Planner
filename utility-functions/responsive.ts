import * as Device from "expo-device";
import { Dimensions } from "react-native";
const {width, height} = Dimensions.get("window");

export const isTablet = () => Device.deviceType === Device.DeviceType.TABLET;

export const isLandscape = ()=>{
    return width > height;
}

export const _isTab=()=>{
    return width > 700 && height > 700 ? true : false;
}