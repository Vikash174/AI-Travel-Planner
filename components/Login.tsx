import { View, Text, Image, StyleSheet, TouchableOpacity } from "react-native";
import React from "react";
import { Colors } from "@/constants/Colors";
import { StatusBar } from "expo-status-bar";
import { isTablet } from "@/utility-functions/responsive";
import { useRouter } from "expo-router";

const isTabletView = isTablet();

export default function Login() {
  const router = useRouter();
  return (
    <View style={styles.container}>
      <StatusBar style="dark" backgroundColor={Colors.white} hidden />
      <Image
        source={require("../assets/images/login-image.jpeg")}
        style={styles.image}
      />
      <View style={styles.bottomSheet}>
        <Text style={styles.title}>AI Travel Planner</Text>
        <Text style={styles.subtitle}>
          Discover your next adventure effortlessly. Personalized itinerary
          plans created by AI.
        </Text>
        <TouchableOpacity
          style={styles.button}
          onPress={() => router.push("/auth/sign-in")}
        >
          <Text style={styles.buttonText}>Get Started</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white,
    width: "100%",
  },
  image: {
    width: "100%",
    height: "70%",
    resizeMode: "cover", // or 'contain', depending on your design
  },
  bottomSheet: {
    flex: 1,
    backgroundColor: Colors.white,
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    padding: 20,
    elevation: 5, // Shadow for Android
    shadowColor: "#000", // Shadow for iOS
    shadowOffset: { width: 0, height: -2 }, // Shadow for iOS
    shadowOpacity: 0.1, // Shadow for iOS
    shadowRadius: 10, // Shadow for iOS
    justifyContent: "space-evenly",
    alignItems: "center",
    flexDirection: "column",
    marginTop: -10,
  },
  title: {
    fontSize: isTabletView ? 30 : 20,
    fontFamily: "outfit-bold",
    textAlign: "center",
    marginBottom: isTabletView ? 20 : 10,
  },
  subtitle: {
    fontSize: isTabletView ? 20 : 15,
    fontFamily: "outfit",
    textAlign: "center",
    marginBottom: 30,
  },
  button: {
    backgroundColor: Colors.primary,
    borderRadius: 30,
    paddingVertical: 15,
    paddingHorizontal: 30,
    alignItems: "center",
    width: isTabletView ? 300 : "100%",
  },
  buttonText: {
    color: Colors.white,
    fontSize: isTabletView ? 20 : 15,
    fontFamily: "outfit",
  },
});
