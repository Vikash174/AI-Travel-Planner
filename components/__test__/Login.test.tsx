import * as Device from "expo-device";
import * as responsive from "@/utility-functions/responsive";
import { render, screen } from "@testing-library/react-native";
import Login from "../Login";

// Mock device functions
jest.mock("@/utility-functions/responsive", () => ({
  isTablet: jest.fn(),
}));

describe("Login component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly on phone", () => {
    jest.spyOn(responsive, "isTablet").mockReturnValue(false);
    const { toJSON } = render(<Login />);

    // check if component renders
    expect(toJSON()).toMatchSnapshot();

    // check if elements are rendered
    expect(screen.getByText("AI Travel Planner")).toBeTruthy();
    expect(
      screen.getByText(/Discover your next adventure effortlessly/i)
    ).toBeTruthy();
    expect(screen.getByText("Sign in with Google")).toBeTruthy();
  });

  test("renders correctly on tablet", () => {
    // Mock isTablet to return true (tablet)

    const { toJSON } = render(<Login />);

    // Check if the component matches the snapshot
    expect(toJSON()).toMatchSnapshot();

    // Check if elements are rendered
    expect(screen.getByText("AI Travel Planner")).toBeTruthy();
    expect(
      screen.getByText(/Discover your next adventure effortlessly/i)
    ).toBeTruthy();
    expect(screen.getByText("Sign in with Google")).toBeTruthy();
  });
});
