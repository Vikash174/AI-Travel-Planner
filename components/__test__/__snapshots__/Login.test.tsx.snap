// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly on phone 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": "70%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 5,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "marginTop": -10,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "outfit-bold",
          "fontSize": 20,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      AI Travel Planner
    </Text>
    <Text
      style={
        {
          "fontFamily": "outfit",
          "fontSize": 15,
          "marginBottom": 30,
          "textAlign": "center",
        }
      }
    >
      Discover your next adventure effortlessly. Personalized itinerary plans created by AI.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0a7ea4",
          "borderRadius": 30,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "outfit",
            "fontSize": 15,
          }
        }
      >
        Sign in with Google
      </Text>
    </View>
  </View>
</View>
`;

exports[`Login component renders correctly on tablet 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": "70%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 5,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "marginTop": -10,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "outfit-bold",
          "fontSize": 20,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      AI Travel Planner
    </Text>
    <Text
      style={
        {
          "fontFamily": "outfit",
          "fontSize": 15,
          "marginBottom": 30,
          "textAlign": "center",
        }
      }
    >
      Discover your next adventure effortlessly. Personalized itinerary plans created by AI.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0a7ea4",
          "borderRadius": 30,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "outfit",
            "fontSize": 15,
          }
        }
      >
        Sign in with Google
      </Text>
    </View>
  </View>
</View>
`;
